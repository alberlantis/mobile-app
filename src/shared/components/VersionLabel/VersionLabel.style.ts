import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    position: "absolute",
    backgroundColor: "transparent",
    bottom: 0,
    left: 0,
    right: 0,
    paddingVertical: 5,
  },
  text: {
    textAlign: "center",
  },
});
