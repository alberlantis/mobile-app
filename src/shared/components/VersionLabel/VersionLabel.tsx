import React from "react";
import { View, Text } from "react-native";
import * as Application from "expo-application";

import { IS_EXPO_GO } from "src/shared/constants/platform";
import {
  EXPO_PUBLIC_APP_VERSION,
  EXPO_PUBLIC_BUILD_VERSION,
} from "src/shared/constants/env";
import s from "./VersionLabel.style";

const VersionLabel = () => {
  const { nativeApplicationVersion, nativeBuildVersion } = Application;
  const version = IS_EXPO_GO
    ? EXPO_PUBLIC_APP_VERSION
    : nativeApplicationVersion;
  const buildNumber = IS_EXPO_GO
    ? EXPO_PUBLIC_BUILD_VERSION
    : nativeBuildVersion;
  return (
    <View style={s.container}>
      <Text style={s.text}>{`${version} (${buildNumber})`}</Text>
    </View>
  );
};

export default VersionLabel;
