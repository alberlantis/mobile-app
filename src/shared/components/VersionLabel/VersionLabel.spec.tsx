import { render, screen } from "@testing-library/react-native";

import VersionLabel from "./VersionLabel";

jest.mock("expo-application", () => ({
  nativeApplicationVersion: "1.5.24",
  nativeBuildVersion: "23",
}));

const platform = {
  IS_EXPO_GO: false,
};
jest.mock("src/shared/constants/platform", () => platform);

jest.mock("src/shared/constants/env", () => ({
  EXPO_PUBLIC_APP_VERSION: "EXPO_PUBLIC_APP_VERSION",
  EXPO_PUBLIC_BUILD_VERSION: "EXPO_PUBLIC_BUILD_VERSION",
}));

describe("VersionLabel", () => {
  describe("when is bare", () => {
    beforeAll(() => {
      platform.IS_EXPO_GO = false;
    });

    beforeEach(() => {
      render(<VersionLabel />);
    });

    it("should match snapshot", () => {
      expect(screen.toJSON()).toMatchSnapshot();
    });

    it("should render correct app version and build number from native code", () => {
      expect(screen.getByText("1.5.24 (23)")).toBeOnTheScreen();
    });
  });

  describe("when is expo go", () => {
    beforeAll(() => {
      platform.IS_EXPO_GO = true;
    });

    beforeEach(() => {
      render(<VersionLabel />);
    });

    it("should match snapshot", () => {
      expect(screen.toJSON()).toMatchSnapshot();
    });

    it("should render correct app version and build number from env variables", () => {
      expect(
        screen.getByText("EXPO_PUBLIC_APP_VERSION (EXPO_PUBLIC_BUILD_VERSION)"),
      ).toBeOnTheScreen();
    });
  });
});
