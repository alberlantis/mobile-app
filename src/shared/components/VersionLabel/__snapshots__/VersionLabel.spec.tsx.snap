// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionLabel when is bare should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "paddingVertical": 5,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <Text
    style={
      {
        "textAlign": "center",
      }
    }
  >
    1.5.24 (23)
  </Text>
</View>
`;

exports[`VersionLabel when is expo go should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "paddingVertical": 5,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <Text
    style={
      {
        "textAlign": "center",
      }
    }
  >
    EXPO_PUBLIC_APP_VERSION (EXPO_PUBLIC_BUILD_VERSION)
  </Text>
</View>
`;
